### PR TITLE
blocking: if you block, delete blocked follower from your profile

### PR DESCRIPTION
**Pull request by maxtaco**
- user.blocked gregor -> notifyUserBlocked notification
- in JS, action to handle that notification -> reducer -> delete blocked users from following Set
- update followingCount appropriately, and maybe find an existing bug

### CODE REVIEW

**Comment by maxtaco** [comment]
r? @keybase/react-hackers 

CC: @zapu

**Comment by jzila** [comment]
Does this also trigger on hide from followers?

File: shared/reducers/tracker2.tsx [hunk 1]
```diff
@@ -106,4 +108,14 @@ export default Container.makeReducer<Actions, Types.State>(initialState, {
       ...rest,
     })
   },
+  [Tracker2Gen.userBlocked]: (draftState, action) => {
```
Comment (chrisnojima) [inline shared/reducers/tracker2.tsx]: You can handle the engine gen action itself and not translate it with a new tracker action plus saga here instead

**Comment by maxtaco** [comment]
@chrisnojima ready for another look

@jzila which flow is that?

File: shared/reducers/tracker2.tsx [hunk 2]
```diff
@@ -106,4 +112,16 @@ export default Container.makeReducer<Actions, Types.State>(initialState, {
       ...rest,
     })
   },
+  [EngineGen.keybase1NotifyTrackingNotifyUserBlocked]: (draftState, action) => {
+    const {blocker, blocked} = action.payload.params.b
+    const d = getDetails(draftState, blocker)
+    const followers = d.followers || new Set()
```
Comment (chrisnojima) [inline shared/reducers/tracker2.tsx]: prefer `??` for fallbacks

File: shared/reducers/tracker2.tsx [hunk 3]
```diff
@@ -106,4 +112,16 @@ export default Container.makeReducer<Actions, Types.State>(initialState, {
...
+    d.followers = followers
+    ;(blocked || []).forEach(e => {
+      followers.delete(e)
+      const dBlocked = getDetails(draftState, e)
```
Comment (chrisnojima) [inline shared/reducers/tracker2.tsx]: can just
`getDetails(draftState, e).blocked = true`

File: shared/reducers/tracker2.tsx [hunk 4]
```diff
@@ -106,4 +112,16 @@ export default Container.makeReducer<Actions, Types.State>(initialState, {
       ...rest,
     })
   },
+  [EngineGen.keybase1NotifyTrackingNotifyUserBlocked]: (draftState, action) => {
+    const {blocker, blocked} = action.payload.params.b
+    const d = getDetails(draftState, blocker)
+    const followers = d.followers || new Set()
+    d.followers = followers
+    ;(blocked || []).forEach(e => {
```
Comment (chrisnojima) [inline shared/reducers/tracker2.tsx]: `const toProcess = blocked ?? []` might be nicer than the leading `;` junk

**Pull request merged**